cmscontrib/loaders/lio.py

```diff
@@ -0,0 +1,322 @@
+            )
+            args["managers"]["checker"] = Manager("checker", digest)
+            evaluation_param = "comparator"
+        else:
+            evaluation_param = "diff"
+
+        point_file = os.path.join(self.task_dir, self.conf.get('point_file', 'punkti.txt'))
+        points_per_group = self.parse_point_file(point_file)
+        if sum(points_per_group.values()) != 100:
+            logger.critical("Points for all groups doesn't sum up to 100")
+            return None
+
+        args["score_type"] = self.conf.get("score_type", "GroupMin")
+        args["score_type_parameters"] = \
+            [[points_per_group[i], f"{i:03}"] for i in range(len(points_per_group))]
+        args["task_type"] = "Batch"
+        args["task_type_parameters"] = \
+            [compilation_param,
+             [input_filename, output_filename],
+             evaluation_param]
+        public_groups = self.conf.get('public_groups', [0, 1])
+
+        args["testcases"] = {}
+        tests_per_group = [0] * len(points_per_group)
+        test_zip = os.path.join(self.task_dir, self.conf.get('test_archive', 'testi.zip'))
+        with zipfile.ZipFile(test_zip) as zip:
+
+            # Collect and organize test files from zip archive
+            matcher = re.compile(r"\.(i|o)(\d+)([a-z]*)$")
+            test_files = {}
+            for test_filename in zip.namelist():
+                if '/' in test_filename or '\\' in test_filename:
+                    logger.critical("Test zip archive contains a directory")
+                    return None
+                match = matcher.search(test_filename)
+                if not match:
+                    logger.critical(f"Unsupported file in test archive {name}")
+                    return None
+
+                is_input = match.group(1) == 'i'
+                group = int(match.group(2))
+                test_in_group = match.group(3)
+
+                if group not in test_files:
+                    test_files[group] = {}
+                if test_in_group not in test_files[group]:
+                    test_files[group][test_in_group] = {}
+
+                testcase = test_files[group][test_in_group]
+                testcase['input' if is_input else 'output'] = test_filename
+
+            # Extract them in correct order and update subtask list and testcases
+            for group in sorted(test_files.keys()):
+                for test_in_group in sorted(test_files[group].keys()):
+                    tests_per_group[group] += 1
+                    testcase = test_files[group][test_in_group]
+                    if 'input' not in testcase or 'output' not in testcase:
+                        logger.critical(f"Input or output file not found for test {group}{test_in_group}")
+                        return None
+                    with zip.open(testcase['input'], 'r') as input_file:
+                        content = io.TextIOWrapper(input_file, encoding='ascii', newline=None).read()
```

Comment:
Here and below we are loading a potentially huge file in memory. Can we only use file objects instead?

Reply:
I wanted to do the new line formatting through the streams and objects before, but I remember that I didn't find how to do that. I remember that  TextIOWrapper didn't support that. Will look into it when I will have time, but as RAM memory is huge nowadays I don't think that these small tests will raise any issue.